### PR TITLE
8963 topics location

### LINKED ISSUE #8963: [Blog] - [Blog Page] - Move Topics location
image

The goal of this ticket is to move the location of the blog pages "topics" from the right side of the author byline, and now rendering them above the blog pages title.

Dev Tasks:
--- 

- [ ] Update the blog page template to render the Topics above the title
- [ ] Update the "Blog_authors" template to no longer render the categories next to the byline

Acceptance Criteria:
--- 

- [ ] Users can select topics for blog posts the same as before
- [ ] The topics now render above the title, and loop through all selected topics, instead of just rendering the first in the list.

### PR DESCRIPTION
**Pull request by tbrlpld**
This moves the location of the "topics" on the blog page from the by-line to above the page title. Since the design is basically the same as the on the blog cards, I use that as an included and only added the option to pass extra classes to the template to account for the larger font when displaying the topics on the blog page. 

To check that two categories work too, add the second category in the admin. 

Closes #8963 
Related PRs/issues #8963

**Link to sample test page**: http://localhost:8000/en/blog/initial-test-blog-post-with-fixed-title/

# Screenshots

## Mobile

**One topic**

**Two topics**

## Desktop

**One topic**

**Two topics**

# Checklist

_Remove unnecessary checks_

**Tests**
- [ ] ~~Is the code I'm adding covered by tests?~~

**Changes in Models:**
- [ ] ~~Did I update or add new fake data?~~
- [ ] ~~Did I squash my migration?~~
- [x] [Are my changes backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- [ ] ~~Is my code documented?~~
- [ ] ~~Did I update the READMEs or wagtail documentation?~~

### CODE REVIEW

**Comment by tbrlpld** [comment]
cc @kevinhowbrook

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/blog_index_featured_video_post.html [hunk 1]
```diff
@@ -3,14 +3,14 @@
   <div class="tw-grid tw-grid-cols-12 tw-mb-7">
     <div class="tw-pr-[36px] tw-py-3 tw-flex tw-flex-col tw-col-span-12 large:tw-col-span-4 large:tw-py-0">
       <div class="tw-flex tw-flex-wrap">
-        {% include "./blog_card_topics.html" with topics=blog_page.specific.topics.all%}
+        {% include "./blog_topics.html" with topics=blog_page.specific.topics.all%}
```
Comment (kevinhowbrook) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/blog_index_featured_video_post.html:6]: Think there is a missing space here before the closing %

```suggestion
        {% include "./blog_topics.html" with topics=blog_page.specific.topics.all %}
```

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/blog_card.html [hunk 2]
```diff
@@ -12,7 +12,7 @@
 each blog page's dominant tag)
 
 {% block tags %}
-  {% include "./blog_card_topics.html" with topics=page.specific.topics.all%}
+  {% include "./blog_topics.html" with topics=page.specific.topics.all%}
```
Comment (kevinhowbrook) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/blog_card.html:15]: Think there is a missing space here before the closing %

```suggestion
  {% include "./blog_topics.html" with topics=page.specific.topics.all %}
```

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/blog_authors.html [hunk 3]
```diff
@@ -7,28 +7,30 @@
...
+
+        {% if show_full_info %}
+          |
+          {% if blog_page.first_published_at %}
+            {{ blog_page.first_published_at|date:"DATE_FORMAT" }}
+          {% else %}
+            not published yet
+          {% endif %}
+        {% endif %}
+      {% endspaceless%}
```
Comment (kevinhowbrook) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/blog_authors.html:33]:
```suggestion
      {% endspaceless %}
```

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/blog_authors.html [hunk 4]
```diff
@@ -7,28 +7,30 @@
...
-      {% endif %}
-      {% endwith %}
-    {% endif %}
-    {% endspaceless%}
+
+        {% for author in authors %}
+          {% if not forloop.first %}{% if not forloop.last %}<span class="tw-font-extrabold">, </span>{% else %}<span> {% trans "and" %} </span>{% endif %}{% endif %}
+          {% if authors_length > 1 and forloop.last %}  {% endif %}
+            {% if blog_index %}
+              <a class="tw-font-extrabold tw-text-black" href="{% routablepageurl page=blog_index url_name='blog-author-detail' profile_slug=author.slug  %}">{{ author.name }}</a>
```
Comment (kevinhowbrook) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/blog_authors.html:19]: Double-space after `author.slug `

```suggestion
              <a class="tw-font-extrabold tw-text-black" href="{% routablepageurl page=blog_index url_name='blog-author-detail' profile_slug=author.slug %}">{{ author.name }}</a>
```

**Review by kevinhowbrook — APPROVED** [review]
Thanks @tbrlpld, just spotted a few formatting things, otherwise looks good :+1:

**Comment by tbrlpld** [comment]
> Thanks @tbrlpld, just spotted a few formatting things, otherwise looks good 👍

Thanks @kevinhowbrook for reviewing and catching all these formatting issues!

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/blog_card.html [hunk 5]
```diff
@@ -12,7 +12,7 @@
 each blog page's dominant tag)
 
 {% block tags %}
-  {% include "./blog_card_topics.html" with topics=page.specific.topics.all %}
+  {% include "./blog_topics.html" with topics=page.specific.topics.all %}
```
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/blog_card.html:15]: **kudos** thanks for the rename! This makes more sense to me and we can share this among other templates

**Review by danielfmiranda — APPROVED** [review]
Hi Tibor! I just pulled this into my local machine and tested it out, everything looks good to me both on the front end and code wise 👍 approved!

**Comment by tbrlpld** [comment]
Thanks for the reviews @kristinashu @danielfmiranda

**Pull request merged**